tests/console/yast2_registration.pm

```diff
@@ -0,0 +1,64 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: check if an unregistered system can be registered and if
+#          enabling and disabling extensions correctly work.
+# Maintainer: Paolo Stivanin <pstivanin@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+
+sub register_system {
```

Comment:
This doesn't work ?
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/registration.pm#L433

Reply:
:thinking: I'm registering and enabling an extension, while that function is just registering the system if I got it right

Reply:
the function is doing complete registration :wink:

Reply:
ok nice :) I'll try it then!!

Reply:
that function isn't working (http://d502.qam.suse.de/tests/1484), I guess I'll stick with my approach :+1:

Reply:
you are missing needle for base module, in the function is already stuff which you maybe don't explicitly need but it's still good to have it, like the missing needle [1]
I think it's better to use this function, sure you can invent wheel with what you need for now, but at the end I'm sure it will be worse than something existing and used for years
[1] https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/registration.pm#L494

Reply:
True, that's a good point. Not using that function would be basically mean reinventing the wheel, thing that we should avoid :+1: 
Ok, I'll create the missing needle(s) and see what's gonna happen :)

Reply:
I've tried this approach, but I had to basically recreate quite some needles and it still fails (http://d502.qam.suse.de/tests/1495).
I'm not really sure this is the best approach :thinking: 
IMHO, I'd prefer having double code (few lines) then lots of almost identical needles to maintain

Reply:
Use your previous code, unfortunately there are variables which affect the function

Reply:
Ok, thank you :)
Now I'm gonna run it again against all SLE versions with some of the suggestions you made!

Reply:
thank you for trying :)